Moggles.Domain/Application.cs

```diff
@@ -116,6 +116,48 @@ public void ChangeEnvironmentDefaultValue(string name, bool newDefaultValue)
             env.DefaultToggleValue = newDefaultValue;
         }
 
+        public void ChangeEnvironmentPosition(string environment, bool moveToLeft, bool moveToRight)
+        {
+            DeploymentEnvironments = DeploymentEnvironments.OrderBy(e => e.SortOrder).ToList();
+            var env = DeploymentEnvironments.FirstOrDefault(e => string.Compare(e.EnvName, environment, StringComparison.OrdinalIgnoreCase) == 0);
+
+            if (env == null)
+                throw new InvalidOperationException("Environment does not exists!");
+
+            if (moveToLeft && !moveToRight)
+            {
+                MoveLeft(env);
+            }
+
+            if(moveToRight && !moveToLeft)
+            {
+                MoveRight(env);
+            }
+        }
+        private void MoveLeft(DeployEnvironment environment)
+        {
+            var index = DeploymentEnvironments.IndexOf(environment);
+            if (index > 0)
```

Comment:
What does this check for? That the environment is not already on the first position?
The check can be extracted in a _Local Function_

Reply:
You can also add a test for this